### ISSUE TITLE
Fine-tune with LORA, high GPU usage

### ISSUE THREAD

**Issue by renxiangnan** [serp-ai/bark-with-voice-clone#42]
Hi guys, thank you for this cool work. I am doing some experiments with fine-tuning on a certain speaker recently. The related parameters are set as:

lora_dim = 64
optimize_lora_params_only = True

However, I've noticed that the GPU memory usage and overall training time remain virtually unchanged, regardless of whether optimize_lora_params_only is set to True or False.

Is this expected? I was under the impression that using LoRa for fine-tuning the model would significantly decrease both GPU memory consumption and training time. Could I be overlooking something? I appreciate your response.

**Comment by aartykov** [comment]
did u try decreasing the lora rank value? It should decrease the memory usage.

**Comment by renxiangnan** [comment]
> did u try decreasing the lora rank value? It should decrease the memory usage.

Thanks for the comment. I tried from 4 to 64, and GPU memo did not change. Any suggestion?

**Comment by aartykov** [comment]
It is highly possible that there is bug in the implementation. Let me run and check the training code.

**Comment by renxiangnan** [comment]
> It is highly possible that there is bug in the implementation. Let me run and check the training code. 

Appreciate that!

**Comment by aartykov** [comment]
Here is the solution: Change torch version from 1.13 to 2 or above.

**Comment by renxiangnan** [comment]
> Here is the solution: Change torch version from 1.13 to 2 or above.

Really appreciate that, issue resolved!

**Comment by aartykov** [comment]
Nice! Looking forward to your training results...

**Comment by renxiangnan** [comment]
> Nice! Looking forward to your training results...

Thanks for your help again. However, I noticed in the load_model method, it looks like all LORA weights have been removed from state dictionary. I just opened an issue here: https://github.com/serp-ai/bark-with-voice-clone/issues/48 . This looks so confused for me. I mean the whole repo aims to fine-tune bark, if lora weight has been removed, what is the purpose then? Any thoughts?

**Comment by aartykov** [comment]
Hey @renxiangnan ! Are you trying finetune the model or ur goal is just voice cloning?

**Comment by renxiangnan** [comment]
> Hey @renxiangnan ! Are you trying finetune the model or ur goal is just voice cloning?

I did fine-tune the model with several target speakers, I would say, the improvement is noticeable especially in terms of timbre.

**Comment by aartykov** [comment]
@renxiangnan Great to hear that the script works. So did you choose the default hyperparameters provided on this repo?

**Comment by renxiangnan** [comment]
@aartykov I have seen the issue you raised, and I did manage to get the loss converge. I did not use the original settings, I tried with lower lora dimension (e.g., 16 or 8), and lr=5e-4. Maybe this will be helpful. However, I also noticed some bad surprise, like the fine-tuned model will generate some random tail speech at the end.

**Comment by aartykov** [comment]
Ohh I see. I think the tail speech can be cropped. Did you also change inference parameters like temperature?

**Comment by renxiangnan** [comment]
@aartykov Thanks for the suggestion. However, I noticed that, for the target speaker with strong accent, fine-tuned model has bigger chance to generate such random tail speech. I did not change the temperature, but that is a good tip. Please let me know if you enable to fine-tune the model.

**Comment by aartykov** [comment]
If you have discord account, please, add me: aartykoff#1142

**Comment by devinschumacher** [comment]
@renxiangnan 

We just launched our follow up to BARK -- an AI Voice Cloner Desktop App (runs locally, offline, super realistic voice cloning, data privacy, no coding needed)

👉 Check it out: https://github.com/serpapps/ai-voice-cloner
